payload/reggae/build.d

```diff
@@ -131,7 +132,7 @@ Target inTopLevelObjDirOf(Target target, string dirName, Flag!"topLevel" isTopLe
     }
 
     auto outputs = isTopLevel
-        ? target._outputs.map!(a => expandBuildDir(a)).array
+        ? target._outputs.map!(a => buildPath(expandBuildDir(a))).array
```

Comment:
I think it's better if `expandBuildDir` always does the right thing instead of expecting callers to know to wrap it in `buildPath`.

And now that I think about it, is it the case that the functions being used from `std.path` aren't doing the right thing on Windows?

Reply:
IIRC, `expandBuildDir` is also used to expand it in arbitrary commands - and I surely don't want to replace `cl.exe /MT` to `cl.exe \MT`.

`std.path.buildPath` works well, but it doesn't replace existing forward slashes in segments, that's why there's `reggae.path.buildPath` instead.

Reply:
Ah.